packages/analytics/analytics-geo-map/src/components/AnalyticsGeoMap.vue

```diff
@@ -0,0 +1,244 @@
+<template>
+  <div
+    id="mapContainer"
+    ref="mapContainer"
+    class="kong-ui-public-analytics-geo-map"
+  />
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, watch, computed } from 'vue'
+import { Map, Popup } from 'maplibre-gl'
+import type { PropType } from 'vue'
+import type { ColorSpecification, DataDrivenPropertyValueSpecification, ExpressionSpecification, LngLatLike, MapOptions } from 'maplibre-gl'
+import type { LongLat, MapFeatureCollection } from '../types'
+import type { FeatureCollection, Feature, MultiPolygon } from 'geojson'
+import 'maplibre-gl/dist/maplibre-gl.css'
+
+const props = defineProps({
+  countryMetrics: {
+    type: Object as PropType<Record<string, number>>,
+    required: true,
+  },
+  geoJsonData: {
+    type: Object as PropType<MapFeatureCollection>,
+    required: true,
+  },
+  center: {
+    type: Object as PropType<LongLat>,
+    required: false,
+    default: null,
+  },
+  fitToCountry: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  initialZoom: {
+    type: Number,
+    required: false,
+    default: null,
+    validator: (value: number) => value >= 0 && value <= 24,
+  },
+  metricUnit: {
+    type: String,
+    required: false,
+    default: '',
+  },
+})
+
+const mapContainer = ref<HTMLDivElement | null>(null)
+const map = ref<Map>()
+
+const layerPaint = computed(() => ({
+  'fill-color': [
+    'match',
+    ['get', 'ISO_A2'] as ExpressionSpecification,
+    ...(Object.keys(props.countryMetrics).length
+      ? Object.entries(props.countryMetrics).flatMap(([code, metric]) => [
+        code,
+        getColor(metric),
+      ])
+      : ['#ccc', '#ccc']),
+    '#ccc', // default color if no match
+  ] as DataDrivenPropertyValueSpecification<ColorSpecification>,
+  'fill-opacity': 0.7,
+}))
+
+const getColor = (metric: number) => {
+  if (metric > 80) return '#296378'
```

Comment:
would a switch be more readable?

Reply:
Changed this completely need a re-review